tests/analysis/multi_delimited_token_stream_tests.cpp

```diff
@@ -0,0 +1,229 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2023 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+#include "analysis/multi_delimited_token_stream.hpp"
+#include "gtest/gtest.h"
+#include "tests_config.hpp"
+
+namespace {
+
+irs::bstring operator""_b(const char* ptr, std::size_t size) {
+  return irs::bstring{
+    irs::ViewCast<irs::byte_type>(std::string_view{ptr, size})};
+}
+
+class multi_delimited_token_stream_tests : public ::testing::Test {
+  virtual void SetUp() {
+    // Code here will be called immediately after the constructor (right before
+    // each test).
+  }
+
+  virtual void TearDown() {
+    // Code here will be called immediately after each test (right before the
+    // destructor).
+  }
+};
+
+}  // namespace
+
+// -----------------------------------------------------------------------------
+// --SECTION--                                                        test suite
+// -----------------------------------------------------------------------------
+
+#ifndef IRESEARCH_DLL
+
+TEST_F(multi_delimited_token_stream_tests, consts) {
+  static_assert("multi-delimiter" ==
+                irs::type<irs::analysis::multi_delimited_token_stream>::name());
+}
+
+TEST_F(multi_delimited_token_stream_tests, test_delimiter) {
+  auto stream =
+    irs::analysis::multi_delimited_token_stream::make({.delimiters = {"a"_b}});
+  ASSERT_EQ(irs::type<irs::analysis::multi_delimited_token_stream>::id(),
+            stream->type());
+
+  ASSERT_TRUE(stream->reset("baccaad"));
+
+  auto* payload = irs::get<irs::payload>(*stream);
```

Comment:
We also need to validate `offset` attribute, And at least in one test `increment` - I know it is constant for this analyzer type but in sake of completeness.